### PR TITLE
basic implementation of feed page

### PR DESCRIPTION
**Pull request by HamdanSCode**

### CODE REVIEW

File: composeCore/build.gradle [hunk 1]
```diff
@@ -49,6 +49,7 @@ dependencies {
     implementation "androidx.compose.ui:ui:$compose_ui_version"
     implementation "androidx.compose.ui:ui-tooling-preview:$compose_ui_version"
     implementation 'androidx.compose.material:material:1.3.1'
+    implementation "androidx.compose.material3:material3:1.0.1"
```
Comment (ForzenAcademy) [inline composeCore/build.gradle]: You shouldn't have both material and material 3

File: createAccountCompose/build.gradle [hunk 2]
```diff
@@ -53,7 +53,7 @@ dependencies {
     implementation 'androidx.activity:activity-compose:1.6.1'
     implementation "androidx.compose.ui:ui:$compose_ui_version"
     implementation "androidx.compose.ui:ui-tooling-preview:$compose_ui_version"
-    implementation 'androidx.compose.material:material:1.3.1'
+    implementation "androidx.compose.material3:material3:1.0.1"
```
Comment (ForzenAcademy) [inline createAccountCompose/build.gradle]: You api this dependency from ComposeCore you wouldn't need to duplicate

File: loginCompose/build.gradle [hunk 3]
```diff
@@ -52,7 +52,7 @@ dependencies {
     implementation "androidx.activity:activity-compose:1.6.1"
     implementation "androidx.compose.ui:ui:$compose_ui_version"
     implementation "androidx.compose.ui:ui-tooling-preview:$compose_ui_version"
-    implementation 'androidx.compose.material:material:1.3.1'
+    implementation "androidx.compose.material3:material3:1.0.1"
```
Comment (ForzenAcademy) [inline loginCompose/build.gradle]: Same

File: mainPageCompose/build.gradle [hunk 4]
```diff
@@ -21,21 +22,37 @@ android {
...
+    implementation project(':mainPageCore')
 
     implementation 'androidx.core:core-ktx:1.9.0'
-    implementation 'androidx.appcompat:appcompat:1.6.1'
-    implementation 'com.google.android.material:material:1.8.0'
+    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.6.1'
+    implementation "androidx.activity:activity-compose:1.7.0"
+    implementation "androidx.compose.ui:ui:1.4.0"
+    implementation "androidx.compose.ui:ui-tooling-preview:1.4.0"
+    implementation "androidx.compose.material3:material3:1.0.1"
```
Comment (ForzenAcademy) [inline mainPageCompose/build.gradle]: Same

File: mainPageCompose/src/main/java/com/hamdan/forzenbook/mainpage/compose/Feed.kt [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                        item.posterLocation,
+                        item.date,
+                        onImageRequestLoad
+                    )
+                    Spacer(
+                        modifier = Modifier
+                            .padding(ForzenbookTheme.dimens.grid.x2)
+                            .background(ForzenbookTheme.colors.colors.primary.copy(alpha = .3f))
+                            .fillMaxWidth()
+                            .height(1.dp)
```
Comment (ForzenAcademy) [inline mainPageCompose/src/main/java/com/hamdan/forzenbook/mainpage/compose/Feed.kt]: Make a `border` dimens/extension

File: mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/database/UserDao.kt [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    @Query(
+        """
+            DELETE FROM ${UserEntity.TABLE_NAME} WHERE ${UserEntity.USER_ID} = :userId
+        """
+    )
+    suspend fun deleteUser(userId: String)
+
+    @Query(
+        """
+            DELETE FROM ${UserEntity.TABLE_NAME} WHERE ${UserEntity.ENTRY_DATE} <= (:currentTime - 86400000)
```
Comment (ForzenAcademy) [inline mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/database/UserDao.kt]: What is this? lol

File: mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/network/UserResponse.kt [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    @SerializedName("location") val location: String,
+    @SerializedName("created_at") val createdAt: String,
+    @SerializedName("avatarUrl") val avatarUrl: String? = null,
+)
+
+fun UserResponse.toUserEntity(): UserEntity {
+    return UserEntity(
+        userId = this.userId,
+        userIcon = this.avatarUrl,
+        name = this.firstName + " " + this.lastName,
```
Comment (ForzenAcademy) [inline mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/network/UserResponse.kt]: Use string template in resources. Asian users would need this localized

File: postCompose/build.gradle [hunk 8]
```diff
@@ -52,7 +52,7 @@ dependencies {
     implementation "androidx.activity:activity-compose:1.7.0"
     implementation "androidx.compose.ui:ui:1.4.0"
     implementation "androidx.compose.ui:ui-tooling-preview:1.4.0"
-    implementation 'androidx.compose.material:material:1.4.0'
+    implementation "androidx.compose.material3:material3:1.0.1"
```
Comment (ForzenAcademy) [inline postCompose/build.gradle]: Same

File: mainPageCompose/src/main/java/com/hamdan/forzenbook/mainpage/compose/Feed.kt (hunk 5 above)
Comment (HamdanSCode) [reply mainPageCompose/src/main/java/com/hamdan/forzenbook/mainpage/compose/Feed.kt]: created

File: mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/database/UserDao.kt (hunk 6 above)
Comment (HamdanSCode) [reply mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/database/UserDao.kt]: this is to delete any old post information whenever a call to retrieve post info is made. It's time is set for 24 hours. it seemed like a cleaner way to just wipe any old posts from room than to keep makeing expensive checks on each post individualy as its asked for.

Comment (ForzenAcademy) [reply mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/database/UserDao.kt]: Yeah but wtf is that ugly af hard coded number. Constant that shiz

Comment (HamdanSCode) [reply mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/database/UserDao.kt]: updated to use a constant.

**Review by ForzenAcademy — APPROVED** [review]

**Pull request merged**